### PR TITLE
Advanced React Sprint

### PR DESCRIPTION
**Pull request by JacobWilliams90**

### CODE REVIEW

**Comment by JacobWilliams90** [comment]
MVP and Stretch completed

File: ANSWERS.md [hunk 1]
```diff
@@ -1,4 +1,18 @@
...
+
 - [ ] What is the purpose of a custom hook?
-- [ ] Why is it important to test our apps?
\ No newline at end of file
+
+    - Allow you to apply non-visual behavior and stateful logic throughout your components by reusing the same hook over and over again. 
+
+- [ ] Why is it important to test our apps?
+
+    - Automated testing minimizes the risk of bugs finding their way into production code.
```
Comment (liam-mei) [inline ANSWERS.md]: I think you're one of the few students that actually used the answer.md file lol.   In particular, side effects were the important part in Class Components.

File: client/src/App.test.js [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+
+it("Dark Mode Button On screen", () => {
+  const { getByText } = rtl.render(<App />);
+  getByText("Dark Mode Toggle");
+});
+
+it("Button toggles and sets Dark Mode", () => {
+  const { getByText, getByTestId } = rtl.render(<App />);
+  rtl.fireEvent.click(getByText("Dark Mode Toggle"));
+  expect(getByTestId("button")).toHaveClass("toggled");
```
Comment (liam-mei) [inline client/src/App.test.js:20]: great job with the click event!

File: client/src/Components/PlayerData.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React, { Component } from "react";
+import axios from "axios";
+import PlayerCard from "./PlayerCard";
+import styled from "styled-components";
+
+const Div = styled.div`
+  display: flex;
```
Comment (liam-mei) [inline client/src/Components/PlayerData.js:7]: Great job with the display: flex!  It's good to continue with these CSS skills!

File: client/src/Components/PlayerData.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  flex-wrap: wrap;
+  justify-content: center;
+`;
+
+export default class PlayerData extends Component {
+  _isMounted = false;
```
Comment (liam-mei) [inline client/src/Components/PlayerData.js:13]: what does this _isMounted dunder do?

Comment (liam-mei) [reply client/src/Components/PlayerData.js:13]: I feel like it can be related to the async nature for the testing?

**Review by liam-mei — APPROVED** [review]

**Comment by liam-mei** [comment]
## MVP
- [x] Answers
- [x] CLASS component to fetch data
- [x] display data
- [x] custom hook
- [x] reasonable amount of test units

## Stretch
- [ ] more custom hooks
- [ ] graphing library

----------
score: ? Need to check for errors and tests
----------
Jacob - Great job with the styling using styled components.  I'm loving the tests you did.  I know this project didn't have much to test, we will be learning a lot more useful testing next testing lesson

**Pull request merged**